src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelEMParams.java

```diff
@@ -0,0 +1,1179 @@
+package org.broadinstitute.hellbender.tools.coveragemodel;
+
+import org.broadinstitute.barclay.argparser.Advanced;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import javax.annotation.Nonnull;
+
+/**
+ * This class represents the hyper-parameters and parameters for {@link CoverageModelEMAlgorithm}
+ *
+ * Works both as an argument collection as well as a getter/setter.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class CoverageModelEMParams {
+
+    public enum PsiUpdateMode {
+        /**
+         * Target-resolved unexplained variance
+         */
+        PSI_TARGET_RESOLVED,
+
+        /**
+         * Isotropic unexplained variance (i.e. one value for all targets)
+         */
+        PSI_ISOTROPIC
+    }
+
+    public enum WSolverStrategy {
+        /**
+         * Perform the M-step update of bias covariates locally
+         */
+        W_SOLVER_LOCAL,
+
+        /**
+         * Perform the M-step update of bias covariates using Spark
+         */
+        W_SOLVER_SPARK
+    }
+
+    public enum CopyRatioHMMType {
+        /**
+         * Update the copy ratio (or copy number) locally
+         */
+        COPY_RATIO_HMM_LOCAL,
+
+        /**
+         * Update the copy ratio (or copy number) using Spark
+         */
+        COPY_RATIO_HMM_SPARK
+    }
+
+    public enum ComputeNodeCommunicationPolicy {
+        /**
+         * Push new data to the compute nodes via broadcast-hash-join method
+         */
+        BROADCAST_HASH_JOIN,
+
+        /**
+         * Make an RDD from the new data and perform a joint-map with the compute note RDDs
+         */
+        RDD_JOIN
+    }
+
+    public static final double DEFAULT_LOG_LIKELIHOOD_TOL = 1e-5;
```

Comment:
Along with removing setters, this should significantly shorten this class.  However, it might still be relatively lengthy simply because you have so many arguments, which introduces another problem---we have to reasonably test and support many combinations of arguments!  I would think carefully about how many knobs you actually want to expose through the command line and what is convenient only for development/debugging.  Then remove knobs or document them as advanced accordingly.